package/gluon-mesh-vpn-wireguard/files/usr/bin/wg-broker-server

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright (C) Christof Schulze <christof@christofschulze.com>. All Rights Reserved.
+#
+
+STARTPORT=40000
+ENDPORT=41000
+MTU=1374
+WG="wg"
+WAN="br-wan"
+
+[[ -f /etc/wg-broker/config ]] && . /etc/wg-broker/config
+
+[[ -f /etc/wg-broker/secret ]] || {
+echo expecting to find wg secret key in /etc/wg-broker/secret >&2
+exit 1
+}
+
+PRIVATEKEY=/etc/wg-broker/secret
+
+set -x
+get_wg_interfaces() {
+    $WG show interfaces |grep babel-wg
+}
+
+get_connection_count() {
+    interfaces=( $(get_wg_interfaces) )
+    echo ${#interface[@]}
+}
+
+get_port() {
+    $WG show ${1:-all} listen-port
+}
+
+port_is_assigned() {
+    get_port | grep -q "$1"
+}
+
+find_free_port() {
+    local port=$((STARTPORT + $RANDOM%1000)) # adding RANDOM to  not assign ports sequentially and savesome time in belows loop
+
+    while $(port_is_assigned $port)
```

Comment:
This is an infinite loop if all 41k ports have been used.

Reply:
indeed. problem is solved now.